### PR TITLE
enable saving run key for ticks that skip runs because of run key idempotence

### PR DESCRIPTION
**Pull request by prha**
## Summary
We only write run_keys when we have submitted runs.  If we write all run keys, we can display which ticks have skipped requests due to idempotence checks.

Required for https://github.com/dagster-io/dagster/pull/7131
Helps address https://github.com/dagster-io/dagster/issues/3726

## Test Plan
BK

### CODE REVIEW

File: python_modules/dagster/dagster/daemon/sensor.py [hunk 1]
```diff
@@ -565,7 +566,6 @@ def _get_or_create_sensor_run(
             # A run already exists and was launched for this time period,
             # but the daemon must have crashed before the tick could be put
             # into a SUCCESS state
-            context.logger.info(f"Skipping run for {run_request.run_key}, found {run.run_id}.")
```
Comment (prha) [inline python_modules/dagster/dagster/daemon/sensor.py:568]: This is optional, but figured that it was spamming the log

File: python_modules/dagster/dagster/daemon/sensor.py [hunk 2]
```diff
@@ -565,7 +566,6 @@ def _get_or_create_sensor_run(
             # A run already exists and was launched for this time period,
             # but the daemon must have crashed before the tick could be put
             # into a SUCCESS state
```
Comment (gibsondan) [inline python_modules/dagster/dagster/daemon/sensor.py:568]: this comment seems very wrong too, remove? Maybe copied over from the scheduler where it is more correct. Unless i'm missing something this happens frequently even if nothing crashed

File: python_modules/dagster/dagster/daemon/sensor.py [hunk 3]
```diff
@@ -89,7 +89,7 @@ def update_state(self, status, **kwargs):
         if origin_run_id:
             self._tick = self._tick.with_origin_run(origin_run_id)
 
-    def add_run(self, run_id, run_key=None):
+    def add_run(self, run_id=None, run_key=None):
```
Comment (gibsondan) [inline python_modules/dagster/dagster/daemon/sensor.py:92]: name has gotten pretty strange - add_run_info?

File: python_modules/dagster/dagster/core/scheduler/instigation.py [hunk 4]
```diff
@@ -497,13 +497,14 @@ def with_status(self, status, error=None, timestamp=None, failure_count=None):
             )
         )
 
-    def with_run(self, run_id, run_key=None):
-        check.str_param(run_id, "run_id")
+    def with_run(self, run_id=None, run_key=None):
```
Comment (gibsondan) [inline python_modules/dagster/dagster/core/scheduler/instigation.py:500]: with_run_info?

**Review by gibsondan — APPROVED** [review]

**Pull request merged**